tests/unit/test_bugjira.py

```diff
@@ -0,0 +1,206 @@
+import jira
+from jira.exceptions import JIRAError
+import bugzilla
+from unittest.mock import Mock, create_autospec
+import pytest
+from xmlrpc.client import Fault
+from pydantic.error_wrappers import ValidationError
+
+from bugjira.broker import Broker, BrokerInitException
+from bugjira.bugjira import Bugjira
+from bugjira.config import get_config
+from bugjira.dao import Issue
+import bugjira.bugzilla as bugjira_bugzilla
+import bugjira.jira as bugjira_jira
+
+
+@pytest.fixture(scope="function", autouse=True)
+def setup(monkeypatch):
+    monkeypatch.setattr(
+        bugjira_bugzilla, "BzBugzilla", create_autospec(bugzilla.Bugzilla)
+    )
+    monkeypatch.setattr(bugjira_jira, "JIRA", create_autospec(jira.JIRA))
+
+
+@pytest.fixture(scope="function")
+def sandboxed_bugjira(good_config_dict):
+    return Bugjira(config=good_config_dict)
+
+
+def test_broker_init_no_backend():
+    """
+    GIVEN the Broker class' constructor
+    WHEN we call it without supplying a backend parameter
+    THEN the resulting instance's backend attribute should be None
+    """
+    broker = Broker(config={})
+    assert broker.backend is None
+
+
+def test_broker_init_with_backend():
+    """
+    GIVEN the Broker class' constructor
+    WHEN we call it with a backend parameter supplied
+    THEN the resulting instance's backend attribute should be the one we input
+        to the constructor
+    """
+    backend = Mock()
+    broker = Broker(backend=backend)
+    assert broker.backend == backend
+
+
+def test_broker_init_with_no_parameters():
+    """
+    GIVEN the Broker class' constructor
+    WHEN we call it with no parameters
+    THEN a BrokerInitException should be raised
+    """
+    with pytest.raises(BrokerInitException):
+        Broker()
+
+
+def test_broker_init_with_no_config_and_no_backend():
+    """
+    GIVEN the Broker class' constructor
+    WHEN we call it with None for both parameters
+    THEN a BrokerInitException should be raised
+    """
+    with pytest.raises(BrokerInitException):
+        Broker(config=None, backend=None)
+
+
+def test_bugjira_init_with_config_dict(good_config_dict):
+    """
+    GIVEN the Bugjira class' constructor
+    WHEN we call it with a valid config dict
+    THEN the resulting Bugjira instance's config attribute will be the dict
+        we input to the constructor
+    """
+    bugjira = Bugjira(config=good_config_dict)
+    assert bugjira.config == good_config_dict
+
+
+def test_bugjira_init_with_config_path(good_config_file_path):
+    """
+    GIVEN the Bugjira class' constructor
+    WHEN we call it with a path to a known good config
+    THEN the resulting instance's config attribute should be equivalent to the
+        contents of the config file
+    """
+    bugjira = Bugjira(config_path=good_config_file_path)
+    config = get_config(config_path=good_config_file_path)
```

Comment:
Yeah, we could do that. My thought was it could be possible for good_config_dict to be defined (in a hypothetical future) as something other than what is in that file, but in reality that's not likely to change. I'll add a comment in the definition of the good_config_dict feature that advises not to change it.